bpf/bpf_overlay.c

```diff
@@ -182,6 +189,10 @@ static inline int handle_ipv4(struct __sk_buff *skb, __u32 *identity)
 		if (unlikely(skb_get_tunnel_key(skb, &key, sizeof(key), 0) < 0))
 			return DROP_NO_TUNNEL_KEY;
 		*identity = key.tunnel_id;
+
+		if (*identity == HOST_ID) {
+			return DROP_INVALID_IDENTITY;
```

Comment:
This is bound to drop some traffic that originates at nodes running older versions, at the point of freshly-upgraded nodes. I presume that this will be fine as users will need to opt-in to this feature during upgrade.

Reply:
This concern is basically addressed in the docs already.